src/algebra/algebra/basic.lean

```diff
@@ -1179,6 +1179,20 @@ instance linear_map.semimodule' (R : Type u) [comm_semiring R]
 
 end algebra
 
+section domain
+
+namespace algebra
+
+variables {R A : Type*} [comm_semiring R] [domain A] [algebra R A]
+
+lemma lmul_injective {x : A} (hx : x ≠ 0) : function.injective (lmul R A x) :=
+(lmul R A x).to_add_monoid_hom.injective_iff.mpr
+  (λ y h, (eq_zero_or_eq_zero_of_mul_eq_zero h).resolve_left hx)
```

Comment:
Can you add this lemma to `algebra/group_with_zero/defs.lean` as  `mul_left_injective'` to match `mul_left_injective`?